jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -579,96 +585,34 @@ def _update_stretch_histogram(self, msg={}):
         if np.any(np.isnan(sub_data)):
             sub_data = sub_data[~np.isnan(sub_data)]
 
-        if self.stretch_histogram is None:
-            # first time the figure is requested, need to build from scratch
-            self.stretch_histogram = bqplot.Figure(padding_y=0)
-
-            hist_x_sc = bqplot.LinearScale()
-            hist_y_sc = bqplot.LinearScale()
-            # TODO: Let user change the number of bins?
-            # TODO: Let user set y-scale to log
-            hist_mark = bqplot.Bins(sample=sub_data, bins=25,
-                                    density=True, colors="gray",
-                                    scales={'x': hist_x_sc,
-                                            'y': hist_y_sc})
-
-            self.stretch_histogram.marks = [hist_mark]
-            self.stretch_histogram.axes = [bqplot.Axis(scale=hist_x_sc,
-                                                       num_ticks=3,
-                                                       tick_format='0.1e',
-                                                       label='pixel value'),
-                                           bqplot.Axis(scale=hist_y_sc,
-                                                       num_ticks=2,
-                                                       orientation='vertical',
-                                                       label='density')]
-
-            self.bqplot_figs_resize = [self.stretch_histogram]
-            self.stretch_hist_nbins = 25
-
-        else:
-            hist_mark = self.stretch_histogram.marks[0]
-            hist_mark.sample = sub_data
-            # TODO: Let user change the number of bins?
-            # TODO: Let user set y-scale to log
+        hist_mark = self.stretch_histogram.marks['histogram']
+        hist_mark.sample = sub_data
+        hist_mark.bins = self.stretch_hist_nbins
+        hist_mark.send_state('sample')
 
         interval = PercentileInterval(95)
         if len(sub_data) > 0:
             hist_lims = interval.get_limits(sub_data)
             hist_mark.min, hist_mark.max = hist_lims
 
-        self._check_if_v_stretch_changed()
-
-    @observe('stretch_vmin_value', 'stretch_vmax_value')
-    def _check_if_v_stretch_changed(self, msg=None):
-        self._remove_histogram_marks()
-        self._add_histogram_marks()
+    @observe('stretch_vmin_value')
+    def _stretch_vmin_changed(self, msg=None):
+        self.stretch_histogram.marks['vmin'].x = [self.stretch_vmin_value] * 2

```

Comment:
No need for multiplication overhead when it is just 2?
```suggestion
        self.stretch_histogram.marks['vmin'].x = [self.stretch_vmin_value, self.stretch_vmin_value]
```